x/ccv/provider/keeper/power_shaping.go

```diff
@@ -0,0 +1,286 @@
+package keeper
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/interchain-security/v5/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v5/x/ccv/types"
+)
+
+// GetConsumerPowerShapingParameters returns the power-shaping parameters associated with this consumer id
+func (k Keeper) GetConsumerPowerShapingParameters(ctx sdk.Context, consumerId string) (types.PowerShapingParameters, error) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ConsumerIdToPowerShapingParametersKey(consumerId))
+	if bz == nil {
+		return types.PowerShapingParameters{}, errorsmod.Wrapf(ccvtypes.ErrStoreKeyNotFound,
+			"GetConsumerPowerShapingParameters, consumerId(%s)", consumerId)
+	}
+	var record types.PowerShapingParameters
+	if err := record.Unmarshal(bz); err != nil {
+		return types.PowerShapingParameters{}, errorsmod.Wrapf(ccvtypes.ErrStoreUnmarshal,
+			"GetConsumerPowerShapingParameters, consumerId(%s): %s", consumerId, err.Error())
+	}
+	return record, nil
+}
+
+// SetConsumerPowerShapingParameters sets the power-shaping parameters associated with this consumer id.
+// Note that it also updates the allowlist and denylist indexes if they are different
+func (k Keeper) SetConsumerPowerShapingParameters(ctx sdk.Context, consumerId string, parameters types.PowerShapingParameters) error {
+	store := ctx.KVStore(k.storeKey)
+	bz, err := parameters.Marshal()
+	if err != nil {
+		return fmt.Errorf("failed to marshal power-shaping parameters (%+v) for consumer id (%s): %w", parameters, consumerId, err)
+	}
+
+	// get old power shaping params
+	oldParameters, err := k.GetConsumerPowerShapingParameters(ctx, consumerId)
```

Comment:
I would recommend just doing:
```
store.Set(...)
k.UpdateAllowlist()
k.UpdateDenylist()
```
I understand that by getting `oldParameters` we gain by not executing the `Update*` methods when not needed, but it seems unnecessary. We do not expect this method to be executed that often (only part of the `MsgCreateConsumer` and `MsgUpdateConsumer`) in any case.

Reply:
What's the concern that you're trying to address? The reason I prefer to only update when it changes is that power shaping params will keep growing. In a permissionless world, these params could change quite often.